### ISSUE TITLE
Weird Apple Script Bug in MacOS Ventura

### ISSUE THREAD

**Issue by campjacob** [ttscoff/bunch#265]
I love Bunch and the work you do!

I'm not sure if this is related to the update or not but I'm assuming it is. I'm also not sure if this might be related, but I was on the beta with no problems, but this started after I upgraded to the offical version. I've started getting an error when I run a couple of my bunches due to "Error running AppleScript"

The following is the consol log.

```
2022-10-27 23:39:42: 🔴{
    NSAppleScriptErrorAppName = Amphetamine;
    NSAppleScriptErrorBriefMessage = "Not authorized to send Apple events to Amphetamine.";
    NSAppleScriptErrorMessage = "Not authorized to send Apple events to Amphetamine.";
    NSAppleScriptErrorNumber = "-1743";
    NSAppleScriptErrorRange = "NSRange: {34, 17}";
}
```

I just have a script that starts and stops the Amphetamine from keeping my mac awake. It used to work, and now it doesn't. The apple script is simple.

%* tell application "Amphetamine" to start new session
!* tell application "Amphetamine" to end session

I built a new bunch with just this and using just the call AppleScript "*" without additional commands.
When I use your sample script `* say "Welcome to the communications context"` it works without problem. I used script editor to just directly run the both of the commands and it runs there. I also created a shortcut to see if it is something related to running it from a third party and also works there. I'm not sure if this is a Bunch problem, Amphetamine problem, or a system program... but I'd love a simple way to tell my Mac to stay awake when I run some of my Bunches.

**Comment by RosemaryOrchard** [comment]
One simple way within Bunch is the caffeinate command! That said, I'd also like to use Amphetamine as I can see it in my menu bar.  Rosemary OrchardOn 28 Oct 2022, at 08:50, Jacob Campbell ***@***.***> wrote:﻿
I love Bunch and the work you do!
I'm not sure if this is related to the update or not but I'm assuming it is. I'm also not sure if this might be related, but I was on the beta with no problems, but this started after I upgraded to the offical version. I've started getting an error when I run a couple of my bunches due to "Error running AppleScript"
The following is the consol log.
2022-10-27 23:39:42: 🔴{
    NSAppleScriptErrorAppName = Amphetamine;
    NSAppleScriptErrorBriefMessage = "Not authorized to send Apple events to Amphetamine.";
    NSAppleScriptErrorMessage = "Not authorized to send Apple events to Amphetamine.";
    NSAppleScriptErrorNumber = "-1743";
    NSAppleScriptErrorRange = "NSRange: {34, 17}";
}

I just have a script that starts and stops the Amphetamine from keeping my mac awake. It used to work, and now it doesn't. The apple script is simple.
%* tell application "Amphetamine" to start new session
!* tell application "Amphetamine" to end session
I built a new bunch with just this and using just the call AppleScript "*" without additional commands.
When I use your sample script * say "Welcome to the communications context" it works without problem. I used script editor to just directly run the both of the commands and it runs there. I also created a shortcut to see if it is something related to running it from a third party and also works there. I'm not sure if this is a Bunch problem, Amphetamine problem, or a system program... but I'd love a simple way to tell my Mac to stay awake when I run some of my Bunches.

**Comment by ttscoff** [comment]
I would take a look at System Settings / Security and ensure that Bunch still has permission to send Apple events to other apps.

Thanks,
Brett

**Comment by ttscoff** [comment]
Oh right, there is always (wake)… but the permissions issue is important to solve anyway.

Thanks,
Brett

**Comment by campjacob** [comment]
Thank you both so much.

I went to System Settings ->  Privacy & Security ->  Automation -> Bunch and I tried toggling the Amphetamine selection a couple of times, but that didn't work. I ended up deleting Bunch and reinstalling and that has seemed to work. When I reinstalled, I installed the beta so it is possible it is that fixed it, but my assumption is that just the deleting and reinstalling fixed whatever permission issues got messed up during the installation.

As a side note, this has been my first time having to really look for things in the new System Settings app, and I must agree about what I feel like I hear everybody in the Apple community talking about. System Settings is not good right now... It was possible to find what I was looking for, but it was not enjoyable one bit to go through and find that settings screen.
